src/backend/graph.test.js

```diff
@@ -87,14 +87,36 @@ describe("graph", () => {
         .addEdge(crabIngredientEdge())
         .addEdge(eatEdge());
 
+    const crabLoop = () => ({
+      address: makeAddress("crab-self-assessment"),
+      src: crabNode().address,
+      dst: crabNode().address,
+      payload: {evaluation: "not effective at avoiding hero"},
+    });
+
+    const critCookEdge = () => ({
+      address: makeAddress("hero_of_time#0@again_cooks@seafood_fruit_mix#3"),
+      src: mealNode().address,
+      dst: heroNode().address,
+      payload: {
+        crit: true,
+        saveScummed: true,
+      },
+    });
+
+    const advancedMealGraph = () =>
+      simpleMealGraph()
+        .addEdge(crabLoop())
+        .addEdge(critCookEdge());
+
     describe("construction", () => {
       it("works for a simple graph", () => {
-        mealGraph();
+        simpleMealGraph();
       });
 
       it("forbids adding an edge with dangling `dst`", () => {
         expect(() => {
-          mealGraph().addEdge({
+          simpleMealGraph().addEdge({
```

Comment:
How about making the "normal test cases" default to working on the advanced graph, unless there's a reason not to?